gtsam/geometry/Pose3.cpp

```diff
@@ -354,6 +354,17 @@ Point3 Pose3::transformFrom(const Point3& point, OptionalJacobian<3, 6> Hself,
   return R_ * point + t_;
 }
 
+Matrix Pose3::transformFrom(const Matrix& points) const {
+  if (points.rows() != 3) {
+    throw std::invalid_argument("Pose3:transformFrom expects 3*N matrix.");
+  }
+  Matrix result(3, points.cols());
+  for (size_t j=0; j < points.cols(); j++) {
+    result.col(j) = transformFrom(Point3(points.col(j)));
+  }
+  return result;
```

Comment:
Why not use homogenous coordinates and make this a simple matrix-matrix multiplication? It will be vectorized and much faster. Makes sense since you're not asking for Jacobians.

Reply:
That's a good point actually. I don't *need* to embed in homogeneous coordinates, I can just do `R*P+t`, although I don't know how to do the '+t' so it vectorizes, so homogeneous might the way to go anyway.